src/bpftrace.cpp

```diff
@@ -682,7 +682,7 @@ int BPFtrace::run_iter()
   }
 
   auto &ap = *attached_probes_.begin();
-  int link_fd = ap->linkfd_;
+  int link_fd = static_cast<AttachedIterProbe *>(ap.get())->iter_link_fd;
```

Comment:
Safer to dynamic_cast here? I think if there's a bug static_cast is UB but dynamic_cast will null ptr exception. cc @amscanne who probably knows the right thing here